src/Readme_irrigation.md

```diff
@@ -0,0 +1,34 @@
+# VIC 4.2.a.irrigation
```

Comment:
@bartnijssen - We haven't fully discussed in the past how we'd like to document these support branches.  Here, @hydrotian has included a README file in the src directory.  I'm wondering if we want to put this documentation somewhere else (either in the header of the root README file or elsewhere).